pkg/metaclient/kvclient/etcdkv/util.go

```diff
@@ -0,0 +1,90 @@
+package etcdkv
+
+import (
+	"strconv"
+
+	"github.com/hanfei1991/microcosm/pkg/metaclient"
+	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/etcdserver/etcdserverpb"
+)
+
+func makePutResp(etcdResp *clientv3.PutResponse) *metaclient.PutResponse {
+	resp := &metaclient.PutResponse{
+		Header: &metaclient.ResponseHeader{
+			// [TODO] use another ClusterID
+			ClusterID: strconv.FormatUint(etcdResp.Header.ClusterId, 10),
+		},
+	}
+
+	return resp
+}
+
+func makeGetResp(etcdResp *clientv3.GetResponse) *metaclient.GetResponse {
+	kvs := make([]*metaclient.KeyValue, 0, len(etcdResp.Kvs))
+	for _, kv := range etcdResp.Kvs {
+		if kv.Version == 0 {
+			// This key has been deleted, don't return to user
```

Comment:
When can this happen?

Reply:
I find the comment on kv.version with `A delete resets the version to zero`.
I will check it on the server side later.